internal/controllers/secretsyncer_controller_test.go

```diff
@@ -0,0 +1,99 @@
+package controllers_test
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/operator-framework/operator-controller/internal/controllers"
+	"github.com/operator-framework/operator-controller/internal/scheme"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestSecretSyncerReconciler(t *testing.T) {
+	secretData := []byte(`{"auths":{"exampleRegistry": "exampledata"}}`)
+	authFile := "/tmp/test-auth.json"
```

Comment:
Don't write to the user's disk this way.

We should use `t.TempDir()` called inside `t.Run()`'s inline function and a relative path for the file from there so that we avoid:
1. Reusing the same path between different test cases
2. Accidentally leaving test files behind on a user's system.